emalloc.c

```diff
@@ -451,13 +451,10 @@ static void* alloc_from_meta(size_t bsize)
     return block_to_payload(b);
 }
 
-int emalloc_init()
+int emalloc_init_reserve(size_t init_size)
```

Comment:
This also changes the behavior of `sgx_mm_init`, i.e., we may need to update the doc here: https://github.com/intel/sgx-emm/blob/222b9d8cd246809cc8da041241824b0d0469000f/README.md?plain=1#L118 if we really decide to go this way.

Reply:
For Intel SDK, this will make some difference. Previously, `sgx_mm_init` will be responsible for reserving some fixed amount of memory but now in the commit this will get gone by the extra call to `emalloc_init_reserve` from tRTS.

For other SDKs, the difference can be ignored because although now we do not reserve memory in `sgx_mm_init` initially, user can still get the memory reserved dynamically when they use the `sgx_mm_xxx` public APIs